### ISSUE TITLE
fhi-aims hamiltonian

### ISSUE THREAD

**Issue by zerothi** [zerothi/sisl#694]
**Describe the feature**

It would be great to allow reading in fhi-aims hamiltonians. In this way we could have an all-electron Hamiltonian access point.

@ialcon here your help could be really valuable, trying to get this up and running requires knowledge on the aims side (which I don't have), and also there is plenty of code in `sisl` to see how to read stuff from.

**Comment by ialcon** [comment]
Yes, this would of course be a fantastic extra tool for `sisl`. I am worried though of a couple of things. First (maybe not that important) is about FHI-AIMS being a private code. Do you know if there would be any issues by implementing these tools without their collaboration/permission? Probably this is not a problem, as `sisl` would only be used for these purposes by FHI-AIMS licensed users after all, but I prefer to get your opinion on this - as far as I recall, you already contacted the FHI-AIMS developing team time ago, but that was not a very successful attempt, right? Secondly, and this worries me a bit more (at least as of know), I have not directly worked with hamiltonian objects from FHI-AIMS - my experience (and that of the regular user of FHI-AIMS, I believe) is based on plotting band-structures and DOS plots (or real-space wave-function distributions, etc) which are directly created from FHI-AIMS via specific parameters within the `control.in` input file. That being said, I will take a look at the manual, as it should be possible to get the hamiltonian printed in some file format.

**Comment by zerothi** [comment]
I have already talked with Volker about this, there should be no issues. :)  
sisl will post-process fhi-aims output. So there should be no licensing issues (as long as one does not copy code from aims).

As for the Hamiltonian output, yeah, that was the main problem. It was not that they weren't interested, however, they of course didn't have time ;)
So if the manual tells something about writing out the Hamiltonian, that would be ideal!

**Comment by ialcon** [comment]
Ok, @zerothi , I will get into it from next Monday/Tuesday. Once I get some useful info from the manual I will let you know, and will start working on it.
